### PR TITLE
stats: fix the index cm sketch for fast analyze

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

The index CM sketch of fast analyze is wrong, it does not consider the prefix columns.

### What is changed and how it works?

Built cm sketch for each prefix columns of index, and merge them into one as the final result.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/analyze.go [hunk 1]
```diff
@@ -986,24 +982,49 @@ func (e *AnalyzeFastExec) buildHist(ID int64, collector *statistics.SampleCollec
...
+		// We need to insert each prefix values into CM Sketch.
+		for i := 0; i < len(idxInfo.Columns); i++ {
+			var err error
+			var value []byte
+			value, remained, err = codec.CutOne(remained)
+			if err != nil {
+				return nil, nil, err
+			}
+			preLen += len(value)
+			data[i] = append(data[i], sample.Value.GetBytes()[:preLen])
```
Comment (winoros) [inline executor/analyze.go]: if it's UTF8 value, we should consider `utf8.RuneCount` instead?

Comment (lamxTyler) [reply executor/analyze.go]: No, it does not matter because every thing is already encoded here.

File: statistics/cmsketch.go [hunk 2]
```diff
@@ -281,14 +281,46 @@ func (c *CMSketch) queryHashValue(h1, h2 uint64) uint64 {
...
+	}
+	sorted := make([]uint64, len(counter))
+	for _, cnt := range counter {
+		sorted = append(sorted, cnt)
+	}
+	sort.Slice(sorted, func(i, j int) bool {
+		return sorted[i] > sorted[j]
+	})
+	numTop = mathutil.MinUint32(uint32(len(counter)), numTop)
+	lastTopCnt := sorted[numTop-1]
```
Comment (crazycs520) [inline statistics/cmsketch.go]: if `numTop` is 0, this will panic?  executor/analyze.go #334 set `numTop` is 0

Comment (lamxTyler) [reply statistics/cmsketch.go]: No, they don't have top n and won't go into this function.

**Pull request merged**